src/canal.erl

```diff
@@ -338,16 +315,15 @@ handle_read_response(
 
     State2 = del_request(RequestId, State),
     gen_server:reply(From, Ret),
-    {noreply, State2}.
+    {noreply, del_request(RequestId, State2)}.
```

Comment:
What is the purpose of this new `del_request` ?

Reply:
I didn't see 2 lines up. State2 has the request deleted.